server/src/test/java/org/elasticsearch/search/profile/query/ProfileCollectorManagerTests.java

```diff
@@ -88,35 +125,127 @@ public void testReduceEmpty() {
+            assertEquals(numDocs, (int) results[1]);
+        }
+        {
+            ProfileCollectorManager<TopDocs> topDocsManager = new ProfileCollectorManager<>(
+                TopScoreDocCollector.createSharedManager(10, null, 1000),
+                "top_docs"
+            );
+            ProfileCollectorManager<Integer> aggsManager = new ProfileCollectorManager<>(
+                new TotalHitCountCollectorManager(),
+                "total_hit_count"
+            );
+            MultiCollectorManager multiCollectorManager = new MultiCollectorManager(topDocsManager, aggsManager);
+            ProfileCollectorManager<Object[]> profileCollectorManager = new ProfileCollectorManager<>(
+                multiCollectorManager,
+                "multi_collector",
+                topDocsManager,
+                aggsManager
+            );
+            Object[] results = searcher.search(new MatchAllDocsQuery(), profileCollectorManager);
+            assertEquals(numDocs, ((TopDocs) results[0]).totalHits.value);
+            assertEquals(numDocs, (int) results[1]);
+            CollectorResult result = profileCollectorManager.getCollectorTree();
+            assertEquals("multi_collector", result.getReason());
+            assertEquals("MultiCollector", result.getName());
+            assertTrue(result.getTime() > 0);
+            assertEquals(2, result.getProfiledChildren().size());
+            ProfilerCollectorResult topDocsCollectorResult = result.getProfiledChildren().get(0);
+            assertEquals("top_docs", topDocsCollectorResult.getReason());
+            assertEquals("SimpleTopScoreDocCollector", topDocsCollectorResult.getName());
+            assertTrue(topDocsCollectorResult.getTime() > 0);
+            ProfilerCollectorResult aggsCollectorResult = result.getProfiledChildren().get(1);
+            assertEquals("total_hit_count", aggsCollectorResult.getReason());
+            assertEquals("TotalHitCountCollector", aggsCollectorResult.getName());
+            assertTrue(aggsCollectorResult.getTime() > 0);
+        }
+    }
+
+    public void testManagerWithSingleChild() throws IOException {
+        {
+            CollectorManager<DummyTotalHitCountCollector, Integer> manager = DummyTotalHitCountCollector.createManager();
+            CollectorManagerWrapper collectorManagerWrapper = new CollectorManagerWrapper(manager);
+            Integer hitCount = searcher.search(new MatchAllDocsQuery(), collectorManagerWrapper);
+            assertEquals(numDocs, hitCount.longValue());
+        }
+        {
+            CollectorManager<DummyTotalHitCountCollector, Integer> manager = DummyTotalHitCountCollector.createManager();
+            ProfileCollectorManager<Integer> profileTopDocsManager = new ProfileCollectorManager<>(manager, "dummy_total_hit_count");
+            CollectorManagerWrapper collectorManagerWrapper = new CollectorManagerWrapper(profileTopDocsManager);
```

Comment:
I tried to understand why this is needed but couldn't figure it out yet. Would you mind giving a short explanation?